lintcheck/src/main.rs

```diff
@@ -342,29 +339,50 @@ impl Crate {
             vec!["--", "--message-format=json", "--"]
         };
 
+        let mut rustc_args = Vec::<&str>::new();
         if let Some(options) = &self.options {
             for opt in options {
-                args.push(opt);
+                rustc_args.push(opt);
             }
         } else {
-            args.extend(&["-Wclippy::pedantic", "-Wclippy::cargo"])
+            rustc_args.extend(&["-Wclippy::pedantic", "-Wclippy::cargo"])
         }
 
         if lint_filter.is_empty() {
-            args.push("--cap-lints=warn");
+            rustc_args.push("--cap-lints=warn");
         } else {
-            args.push("--cap-lints=allow");
-            args.extend(lint_filter.iter().map(|filter| filter.as_str()))
+            rustc_args.push("--cap-lints=allow");
+            rustc_args.extend(lint_filter.iter().map(|filter| filter.as_str()))
+        }
+
+        if let Some(server) = server {
+            let target = shared_target_dir.join("recursive");
+
+            let status = Command::new("cargo")
+                .arg("check")
+                .arg("--quiet")
+                .current_dir(&self.path)
+                .env("CLIPPY_ARGS", rustc_args.join("__CLIPPY_HACKERY__"))
```

Comment:
aren't these clippy args technically?  Perhaps rename to `clippy_args`? :thinking:

Reply:
Yeah, that's true